### ISSUE TITLE
Does it support sniffing opened ports?

### ISSUE THREAD

**Issue by Grabber** [sirgal/Port-Sniffer#2]
@sirgal, does it support sniffing data from already opened ports on Windows?

**Comment by sirgal** [comment]
@Grabber no, it uses Qt built-in serial port lib that can't do that. But it can open two ports and pass-through data from one port into another, while sniffing.

**Comment by Grabber** [comment]
@sirgal 

Could you please explain what are "channel" and "retranslator" options?

I understood it is possible to open a physical device at PORT_X and forward data to PORT_Y, acting as MITM (Man-In-The-Middle)?

**Comment by sirgal** [comment]
@Grabber sorry for misinforming, I didn't actually finish that feature. I've hacked it up real quick, without error checking etc: d97b7b1
Just confirmed it working on my 3d printer: add a virtual com port pair using 3rd party software (I've used https://www.eltima.com/products/vspdxp/ ), setup retranslator from the physical port into one of the virtual ones, then connect your software to the second virtual port.

**Comment by sirgal** [comment]
@Grabber btw, this repo is my old student project, it was made as a tool for one specific task and is meant to be built upon, not simply used as is. I don't even remember the parsing language, have to reverse-engineer it :(
What do you plan using it for?

**Comment by Grabber** [comment]
@sirgal 

I have one customer that use this barcode scanner (https://www.access-is.com/products/bgr135) and I am trying to find a way to keep the barcode reading software working while also having access to the raw data. The reading software is from a third-party.

**Comment by sirgal** [comment]
@Grabber oh, cool! Yeah, it can act like that. I've added live timeline update, check the latest commit. You can't do much with the data though, unless you reverse-engineer the parsing language or add a button to save ```data_holder``` contents somewhere.

**Comment by Grabber** [comment]
@sirgal what is your email, skype or whatsapp? I have a proposal for you!

**Comment by sirgal** [comment]
@Grabber write your ideas here, I'll do them if I have a free time :)
btw, found an example of the program on the parsing language: https://github.com/sirgal/Port-Sniffer/blob/master/parser/testcases.h#L487
